File: benchmarks/fair/strcat.js
const n = Number(process.env.BENCH_N) > 0 ? Number(process.env.BENCH_N) : 2000000;
const parts = [];
for (let i = 0; i < n; i++) { parts.push(String(i % 1000)); parts.push(","); }
const s = parts.join("");
let cnt = 0;
for (let i = 0; i < s.length; i++) if (s.charCodeAt(i) === 44) cnt++;
console.log(s.length + " " + cnt);

File: benchmarks/fair/strcat.py
import os
n = int(os.environ.get("BENCH_N") or 0) or 2000000
parts = []
for i in range(n):
    parts.append(str(i % 1000))
    parts.append(",")
s = "".join(parts)
print(str(len(s)) + " " + str(s.count(",")))
